Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs

```diff
@@ -1023,6 +1032,8 @@ public override void Copy(MSTSWagon copy)
             WaterScoopDepthM = locoCopy.WaterScoopDepthM;
             WaterScoopWidthM = locoCopy.WaterScoopWidthM;
             MoveParamsToAxle();
+            if (locoCopy.CruiseControl != null)
+                CruiseControl = locoCopy.CruiseControl;
```

Comment:
I believe you have to perform a deep clone of the CruiseControl class, not just copying the reference. The Copy() method is called when there are several instances of the same engine in the consist, to avoid parsing the STF file more than once.

Reply:
Will do Copy() method for the CruisControll class. What variables should the copy method include?

Reply:
The Copy() method is a substitution for Parse(STFReader stf), so you should set the same variables as in the CruiseControl.Parse() method. For example, UseThrottleAsSpeedSelector, MaxAccelerationMpSS, etc.